components/mosart/src/riverroute/RtmMod.F90

```diff
@@ -4183,6 +4286,17 @@ subroutine MOSART_init
               TUnit%hlen(iunit) = hlen_max   ! allievate the outlier in drainage density estimation. TO DO
            end if
            rlen_min = sqrt(TUnit%area(iunit))
+           if(TUnit%rlen(iunit) < rlen_min .and. TUnit%mask(iunit)==1) then
```

Comment:
@liho745 @bishtgautam This is where I think may caused changes compared to BASELINE simulation.

Reply:
I can't recall whether I've changed these lines for MOSART-sediment. @donghuix in your tests, is there something for BFB?

Reply:
I just run the `e3sm_mosart_developer` test. The test will try to compare the results form the new rebase branch to a baseline branch. As BASELINE comparison failed, it means we got a different simulation in the new rebase branch.

Reply:
Could you comment off these lines (4299-4308) and run the test again to see whether a BASELINE comparison passes? These lines can be further refined, but they help improve numerical stability in the routing process.